### PR TITLE
[DYN-3117] UnitsUI Tests

### PR DESCRIPTION
**Pull request by Astul-Betizagasti**
### Purpose

Increase test coverage for the Libraries/UnitsUI folder

### Declarations

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Reviewers

Aaron Tang (@QilongTang )
Michael Kirschner (@mjkkirschner )
Aparajit Pratap (@aparajit-pratap )

### FYIs

Alfredo Pozo (@alfredo-pozo )

### CODE REVIEW

File: test/DynamoCoreWpfTests/UnitsUITests.cs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+            Open(@"UI\UnitsUITests.dyn");
+            ViewModel.HomeSpace.RunSettings.RunType = RunType.Manual;
+
+            Run();
+            var node = Model.CurrentWorkspace.NodeFromWorkspace<LengthFromString>("5705381c277c4a86bf6650aeda12a468");
+            Assert.IsFalse(node.IsModified);
+
+            Model.PreferenceSettings.NumberFormat = "0.0";
+
+            node = Model.CurrentWorkspace.NodeFromWorkspace<LengthFromString>("5705381c277c4a86bf6650aeda12a468");
```
Comment (aparajit-pratap) [inline test/DynamoCoreWpfTests/UnitsUITests.cs:92]: Why do you have to obtain the `node` again? It's already assigned in line 87?

**Review by aparajit-pratap — COMMENTED** [review]
What about tests for `AreaFromString` and `VolumeFromString`?

File: test/DynamoCoreWpfTests/UnitsUITests.cs (hunk 1 above)
Comment (QilongTang) [reply test/DynamoCoreWpfTests/UnitsUITests.cs:92]: Echo that, copy pasted maybe?

**Review by QilongTang — APPROVED** [review]
LGTM with one comment to address by @aparajit-pratap

File: test/DynamoCoreWpfTests/UnitsUITests.cs (hunk 1 above)
Comment (Astul-Betizagasti) [reply test/DynamoCoreWpfTests/UnitsUITests.cs:92]: Those classes where already covered, missing only one constructor on each. Also, the constructors on those classes have an obsolete warning. Of the three implementations of `MeasurementInputBase`, `LengthFromString` is the only one without that warning.

https://github.com/DynamoDS/Dynamo/blob/f39159b46625049974a510a043543fababbf51f0/src/Libraries/UnitsUI/UnitsUI.cs#L328-L332

https://github.com/DynamoDS/Dynamo/blob/f39159b46625049974a510a043543fababbf51f0/src/Libraries/UnitsUI/UnitsUI.cs#L288-L292

Comment (QilongTang) [reply test/DynamoCoreWpfTests/UnitsUITests.cs:92]: @Astul-Betizagasti I think to @aparajit-pratap 's point here, can we skip this line and use node you got from Line 87?

Comment (astulb) [reply test/DynamoCoreWpfTests/UnitsUITests.cs:92]: Oh sure, for some reason I didn't see that comment before, I'll change that

Comment (Astul-Betizagasti) [reply test/DynamoCoreWpfTests/UnitsUITests.cs:92]: Oh sure, I didn't see that comment before for some reason, I will change that now and update

**Pull request merged**